src/factories/routePathFactory.ts

```diff
@@ -50,6 +50,7 @@ class RoutePathFactory {
 
         return {
             routePath,
+            // TODO Determine duplicates by coordinates.
```

Comment:
Why this TODO comment? Can node's have the same ID?

I think there can be a case where 2 nodes could be on top of each other so determing by coordinates might not work either